lib/instrumentation/modules/mongodb.js

```diff
@@ -21,10 +21,16 @@ module.exports = (mongodb, agent, { version, enabled }) => {
     listener.on('succeeded', onEnd)
     listener.on('failed', onEnd)
   } else if (mongodb.MongoClient) {
-    // mongo 4.0+ removed the instrument() method in favor of
+    // mongodb 4.0+ removed the instrument() method in favor of
     // listeners on the instantiated client objects.
     class MongoClient extends mongodb.MongoClient {
       constructor () {
+        // The `command*` events are only sent if `monitorCommands: true`.
+        if (!arguments[1]) {
+          arguments[1] = { monitorCommands: true }
+        } else if (arguments[1].monitorCommands !== true) {
+          arguments[1] = Object.assign({}, arguments[1], { monitorCommands: true })
+        }
```

Comment:
If I'm reading this right, it looks like we're removing the ability for an end user to disable the `monitorCommands` events entirely.  This means if a user wanted to disable the mongodb events for reasons (perf., etc.) but still take advantage of other agent features they couldn't.  

Do you think it's worth/possible to handle the above use case by keeping the `if (!arguments[1]) {` conditional, but dropping the `else if (arguments[1].monitorCommands !== true) {` and honoring explict requests to turn off the command events?  Or would that complicate things?

Reply:
> This means if a user wanted to disable the mongodb events for reasons (perf., etc.) but still take advantage of other agent features they couldn't.

They could still set `disableInstrumentations=mongodb` config to the APM agent to disable.

* * *

It would certainly be possible to honour an explicit `arguments[1].monitorCommands === false`. That would allow a (savvy) user to be able to have APM instrumentation on one `MongoClient()` instance and not on another, if they had a use case for that. 

The *fabricated and unlikely* "use case" that will be broken with this change would be the ops person deploying an app for which they want mongodb APM tracing, but the app code has a lingering `{monitorCommands: false}` for whatever reason.

The mongodb code (nothing mentioned in the docs): https://github.com/mongodb/node-mongodb-native/blob/4ecaa37f72040ed8ace6eebc861b43ee9cb32a99/src/mongo_client.ts#L369-L378 says "Intended for APM use only" for the accessors of this config option. That slightly implies it is unexpected that an end user would be fiddling with this config option.

I'm slightly biased against adding the subtlety here, but not strongly. I'm happy to make the change (to honour `arguments[1].monitorCommands === false`) if you prefer. Thoughts?